samza-core/src/main/scala/org/apache/samza/job/local/ThreadJobFactory.scala

```diff
@@ -54,18 +52,31 @@ class ThreadJobFactory extends StreamJobFactory with Logging {
       case _ => None
     }
 
+    val containerListener = new SamzaContainerListener {
+      override def onContainerFailed(t: Throwable): Unit = {
+        throw t
```

Comment:
I think the behavior should be same as before. ThreadJob does have a try-catch and it re-throws the Throwable. Not sure if the JobRunner will handle it though. There isn't any uncaught exception handler in thread job. So, may be ApplicationRunner should deal with it??